CelesteNet.Client/Components/CelesteNetPico8Component.cs

```diff
@@ -0,0 +1,467 @@
+    {
+        if (state.SID != "PICO-8") return;
+        
+        player.Location.Icon = "menu/pico8";
+        player.Location.Side = "";
+        player.Location.TitleColor = new Color(0xff, 0xf1, 0xe8); // PICO-8 White
+    }
+
+    private void OnPlayerSpawn(On.Celeste.Pico8.Classic.player_spawn.orig_ctor orig, Classic.player_spawn self)
+    {
+        orig(self);
+        _inGame = true;
+        _alive = true;
+        Context.Main.SendState();
+    }
+
+    private void OnEmulatorUpdate(On.Celeste.Pico8.Emulator.orig_Update orig, Emulator self)
+    {
+        if (!Settings.Connected) Ghosts.Clear();
+
+        if (!(Context?.Chat?.Active ?? false)) orig(self);
+    }
+
+    private void OnPlayerDraw(On.Celeste.Pico8.Classic.player.orig_draw orig, Classic.player self)
+    {
+        lock (Ghosts)
+            foreach (var ghost in Ghosts.Values)
+                ghost.draw();
+
+        orig(self);
+        if (!Settings.InGameHUD.ShowOwnName) return;
+        
+        var name = Client?.PlayerInfo?.DisplayName ?? "You";
+        var colonIndex = name.LastIndexOf(':');
+        if (colonIndex > -1)
+            name = name[(colonIndex + 2)..];
+        Pico8FontHelper.PrintOutlinedCenter(
+            name, (int) self.x + 4, (int) self.y - 8
+        );
+    }
+
+    private void OnEmulatorReset(On.Celeste.Pico8.Emulator.orig_ResetScreen orig, Emulator self)
+    {
+        Logger.Log(LogLevel.DBG, "PICO8-CNET", $"EMULATOR RESET");
+        _alive = false;
+        _inGame = false;
+        Client?.Send(new DataPicoFrame() {Player = Client?.PlayerInfo, Level = -1});
+        orig(self);
+        _classic = null;
+        _classicData = null;
+        _emulatorData = null;
+        Context.Main.SendState();
+    }
+
+    private void OnPlayerKill(On.Celeste.Pico8.Classic.orig_kill_player orig, Classic self, Classic.player obj)
+    {
+        Logger.Log(LogLevel.DBG, "PICO8-CNET", $"PLAYER DEAD");
+        _alive = false;
+        Client?.Send(new DataPicoFrame() {Player = Client?.PlayerInfo, Dead = true});
+        orig(self, obj);
+    }
+
+    private void OnDrawHair(On.Celeste.Pico8.Classic.player_hair.orig_draw_hair orig, Classic.player_hair self, Classic.ClassicObject obj, int facing, int djump)
+    {
+        orig(self, obj, facing, djump);
+
+        // FIXME: This kinda sucks. Making a new DynamicData 7 times an update is bad for performance.
+        #nullable disable
+        DynamicData data = new(self);
+        // This type is private :/
+        object[] hairNodes = (object[]) data.Get("hair");
+        if (hairNodes == null) return;
+
+        _queuedGhostFrame.Hair.Clear();
+        foreach (object node in hairNodes) {
+            DynamicData nodeData = new(node);
+
+            var x = (float?) nodeData.Get("x") ?? float.NaN;
+            var y = (float?) nodeData.Get("y") ?? float.NaN;
+            var size = (float?) nodeData.Get("size") ?? float.NaN;
```

Comment:
use `.Get<float>()`? 🤔 
Unsure why VS then tells me it wouldn't even return a ` float?` but maybe that's nullable-context jank we gotta fix...